### PR TITLE
scan: implement ui and wireup redelegate modal

### LINKED ISSUE #1968: Implement UI + wire up for sending reledegate msg on modal
Use `react-select` for dropdown.

https://react-select.com/home

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1968

### CODE REVIEW

File: scan/src/components/modal/submitTx/RedelegateMsg.re [hunk 1]
```diff
@@ -1,56 +1,181 @@
...
+    {switch (allSub) {
+     | Data((_, _, {amount: stakedAmount})) =>
+       let maxValInUband = stakedAmount |> Coin.getUBandAmountFromCoin;
+       <EnhanceTxInput
+         width=300
+         inputData=amount
+         setInputData=setAmount
+         parse={Parse.getBandAmount(maxValInUband)}
+         maxValue={maxValInUband /. 1e6 |> Js.Float.toString}
+         msg="Undelegate Amount (BAND)"
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/RedelegateMsg.re]: Let's use Amount

File: scan/src/components/modal/submitTx/RedelegateMsg.re [hunk 2]
```diff
@@ -1,56 +1,181 @@
...
+         placeholder="Insert unbonding amount"
+         inputType="number"
+         code=true
+         autoFocus=true
+       />;
+     | _ => <EnhanceTxInput.Loading msg="Undelegate Amount (BAND)" width=300 />
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/RedelegateMsg.re]: Let's use Amount

File: scan/src/components/modal/ValidatorStakingInfo.re [hunk 3]
```diff
@@ -197,6 +199,13 @@ module StakingInfo = {
             <Text value="Undelegate" />
           </button>
           <HSpacing size=Spacing.md />
+          <button
+            className={Styles.button(100)}
+            onClick={_ => {redelegate()}}
+            disabled={balanceAtStakeAmount.amount == 0. || isReachUnbondingLimit}>
```
Comment (evilpeach) [inline scan/src/components/modal/ValidatorStakingInfo.re]: I don't think we can use `isReachUnbondingLimit`.

File: scan/src/components/modal/submitTx/RedelegateMsg.re [hunk 4]
```diff
@@ -1,56 +1,181 @@
...
+      );
+    let validatorList =
+      filteredValidators->Belt_Array.map(({operatorAddress, moniker}) =>
+        ReactSelect.{value: operatorAddress |> Address.toBech32, label: moniker}
+      );
+    <ReactSelect
+      options=validatorList
+      onChange={newOption => {
+        let newVal = newOption;
+        setSelectedValidator(_ => newVal);
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/RedelegateMsg.re:58]: Add `TODO: Hack styles for react-select` here.

**Review by evilpeach — CHANGES_REQUESTED** [review]
1. Let's add the limit to 7 warning msg like undelegate msg modal.
2. Add more gas 
image

File: scan/src/components/modal/submitTx/RedelegateMsg.re [hunk 5]
```diff
@@ -1,56 +1,192 @@
...
+     | Data((_, _, {amount: stakedAmount})) =>
+       let maxValInUband = stakedAmount |> Coin.getUBandAmountFromCoin;
+       <EnhanceTxInput
+         width=300
+         inputData=amount
+         setInputData=setAmount
+         parse={Parse.getBandAmount(maxValInUband)}
+         maxValue={maxValInUband /. 1e6 |> Js.Float.toString}
+         msg="Amount (BAND)"
+         placeholder="Insert unbonding amount"
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/RedelegateMsg.re]: It should be `redelegate amount` or `amount`.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**